lib/Runtime/Library/JsBuiltInEngineInterfaceExtensionObject.cpp

```diff
@@ -224,14 +224,35 @@ namespace Js
         }
     }
 
-    DynamicObject* JsBuiltInEngineInterfaceExtensionObject::GetPrototypeFromName(Js::PropertyIds propertyId, ScriptContext* scriptContext)
+    DynamicObject* JsBuiltInEngineInterfaceExtensionObject::GetPrototypeFromName(Js::PropertyIds propertyId, BOOL staticMethod, ScriptContext* scriptContext)
     {
         JavascriptLibrary* library = scriptContext->GetLibrary();
 
+        if (staticMethod)
+        {
+            switch (propertyId) {
+            case PropertyIds::Array:
+                return library->arrayConstructor;
+
+            case PropertyIds::Object:
+                return library->objectConstructor;
+
+            case PropertyIds::String:
+                return library->stringConstructor;
+
+            default:
+                AssertMsg(false, "Unable to find a constructor that match with this className.");
```

Comment:
I would AssertOrFailFast for any cost we never expect to run, since the extra if check should never actually affect runtime performance.

Reply:
I was copying code below that just did an Assert for this - relevant point though - should only come up when developing a new builtin and a silent fail isn't great.